client/sites-dashboard/components/link-in-bio-banner/use-link-in-bio-banner.tsx

```diff
@@ -0,0 +1,57 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { Banners } from 'calypso/sites-dashboard/components/link-in-bio-banner/Banners';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference } from 'calypso/state/preferences/selectors';
+import { BannerType } from './BannerType';
+
+const preference = `link-in-bio-banner`;
+
+export const useLinkInBioBanner = () => {
+	const dispatch = useDispatch();
+	const isMobile = useMobileBreakpoint();
+	const isBannerVisible = useSelector( ( state ) => getPreference( state, preference ) );
+	const showBanner = isBannerVisible === undefined || isBannerVisible;
+
+	const getBanner = ( sites: SiteExcerptData[], displayMode: 'row' | 'grid' ) => {
+		let bannerType = BannerType.None;
+		if ( showBanner ) {
+			if ( isMobile ) {
+				bannerType = BannerType.Tile;
+			} else if ( displayMode === 'row' && sites.length === 1 ) {
+				bannerType = BannerType.Row;
+			} else if ( sites.length === 1 ) {
+				bannerType = BannerType.DoubleTile;
+			} else if ( sites.length === 2 ) {
+				bannerType = BannerType.Tile;
+			}
+		}
+		return Banners[ bannerType ];
+	};
+
+	// TODO just for testing the PR - will be removed.
```

Comment:
Another handy way to reset preferences during development is to delete it using the Dev menu.

![CleanShot 2022-10-03 at 17 32 44@2x](https://user-images.githubusercontent.com/1500769/193502216-9bcb0270-a4e4-40a9-9a5c-1c84c8822f78.png)

Reply:
Noted, thanks.